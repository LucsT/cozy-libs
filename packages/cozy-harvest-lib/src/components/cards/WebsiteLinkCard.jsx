import React from 'react'
import PropTypes from 'prop-types'

import Card from 'cozy-ui/transpiled/react/Card'
import { SubTitle, Caption } from 'cozy-ui/transpiled/react/Text'
import { ButtonLink } from 'cozy-ui/transpiled/react/Button'
import { Media, Img, Bd } from 'cozy-ui/transpiled/react/Media'
import Icon from 'cozy-ui/transpiled/react/Icon'
import palette from 'cozy-ui/transpiled/react/palette'

import { translate } from 'cozy-ui/transpiled/react/I18n'

const WebsiteLinkCard = ({ link, t }) => {
  const linkStyle = { textTransform: 'lowercase' }
  const label = new URL(link).host

  return (
    <Card>
      <SubTitle className="u-mb-1">{t('card.websiteLink.title')}</SubTitle>
      <hr className="u-silver" />
      <Media align="top">
        <Img className="u-pr-1">
          <Icon icon="globe" color={palette['coolGrey']} />
        </Img>
        <Bd>
          <ButtonLink
            subtle
            target="_blank"
            href={link}
            label={label}
            theme="text"
            className="u-dodgerBlue u-m-0"
            style={linkStyle}
          />
          <Caption>{t('card.websiteLink.description')}</Caption>
        </Bd>
      </Media>
    </Card>
  )
}

WebsiteLinkCard.propTypes = {
  link: PropTypes.string.isRequired
}

export default translate()(WebsiteLinkCard)
