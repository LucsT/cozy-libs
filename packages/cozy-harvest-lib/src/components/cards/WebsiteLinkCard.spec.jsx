import React from 'react'
import { shallow } from 'enzyme'
import I18n from 'cozy-ui/transpiled/react/I18n'

import WebsiteLinkCard from './WebsiteLinkCard'
import enLocale from '../../locales/en'

describe('WebsiteLinkCard', () => {
  it('should render', () => {
    const wrapper = shallow(
      <I18n dictRequire={() => enLocale} lang="en">
        <WebsiteLinkCard link="https://fc.assure.ameli.fr/FRCO-app/login" />
      </I18n>
    )
    const component = wrapper.shallow()
    expect(component.getElement()).toMatchSnapshot()
  })
})
