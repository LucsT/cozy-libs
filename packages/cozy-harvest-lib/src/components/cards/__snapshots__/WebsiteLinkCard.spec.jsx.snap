// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebsiteLinkCard should render 1`] = `
<withI18n(WebsiteLinkCard)
  link="https://fc.assure.ameli.fr/FRCO-app/login"
/>
`;
